### ISSUE TITLE
JqwikStringSupport accesses ClassUtils illegally

### ISSUE THREAD

**Issue by sormuras** [jlink/jqwik#59]
### Description

JUnit 5.5 RC1 ships with compiled Java module descriptors hiding internal packages, such as `org.junit.platform.commons.util`.

### Stacktrace
```
WARNING: TestEngine with ID 'jqwik' failed to discover tests
java.lang.IllegalAccessError:
class net.jqwik.engine.support.JqwikStringSupport (in module net.jqwik.engine)
cannot access class org.junit.platform.commons.util.ClassUtils (in module org.junit.platform.commons)
because module org.junit.platform.commons does not export org.junit.platform.commons.util to module net.jqwik.engine
        at net.jqwik.engine@1.1.5/net.jqwik.engine.support.JqwikStringSupport.parameterTypesToString(JqwikStringSupport.java:10)
...
        at org.junit.platform.launcher@1.5.0-RC1/org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:177)
        at org.junit.platform.launcher@1.5.0-RC1/org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:164)
...
```

### Work-around
Pass the following option at test runtime:
`--add-opens org.junit.platform.commons/org.junit.platform.commons.util=net.jqwik.engine`

**Comment by jlink** [comment]
Thanks for the heads up. Which Java version produces the exception?

Your suggested workaround would require that everybody using jqwik will add the runtime option, right?

**Comment by sormuras** [comment]
I ran Java 11, iirc. But all Java 9+ runtimes are affected.

> Your suggested workaround would require that everybody using jqwik will add the runtime option, right?

Yes. But only when running (tests) on the module path. Thus, almost(!) nobody as of today.

**Comment by marcphilipp** [comment]
Which `ClassUtils` methods are used?

**Comment by sormuras** [comment]
https://github.com/jlink/jqwik/blob/master/engine/src/main/java/net/jqwik/engine/support/JqwikStringSupport.java#L10

`ClassUtils.nullSafeToString(Class<?>...)`

**Comment by marcphilipp** [comment]
Looks like it's just `nullSafeToString` which is available as a public API via `ClassSupport` since JUnit Platform 1.1.

**Comment by jlink** [comment]
I just removed all dependencies on ClassUtils and StringUtils - and hopefully all other junit internal classes.

**Comment by jlink** [comment]
Will be available in 1.1.6
